When state persistence is enabled as true, then siddhi containing the rabbitmq as source fails to deploy

### Description

When state persistence is enabled as true, then siddhi containing the rabbitmq as source fails to deploy.

 ERROR {org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorDeployer} - java.lang.NullPointerException org.wso2.carbon.streaming.integrator.core.internal.exception.SiddhiAppDeploymentException: java.lang.NullPointerException
        at org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorDeployer.deploySiddhiQLFile(StreamProcessorDeployer.java:132)
        at org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorDeployer.deploy(StreamProcessorDeployer.java:358)
        at org.wso2.carbon.deployment.engine.internal.DeploymentEngine.lambda$deployArtifacts$0(DeploymentEngine.java:291)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at org.wso2.carbon.deployment.engine.internal.DeploymentEngine.deployArtifacts(DeploymentEngine.java:282)
        at org.wso2.carbon.deployment.engine.internal.RepositoryScanner.sweep(RepositoryScanner.java:112)
        at org.wso2.carbon.deployment.engine.internal.RepositoryScanner.scan(RepositoryScanner.java:68)
        at org.wso2.carbon.deployment.engine.internal.DeploymentEngine.start(DeploymentEngine.java:121)
        at org.wso2.carbon.deployment.engine.internal.DeploymentEngineListenerComponent.onAllRequiredCapabilitiesAvailable(DeploymentEngineListenerComponent.java:216)
        at org.wso2.carbon.kernel.internal.startupresolver.StartupComponentManager.lambda$notifySatisfiableComponents$7(StartupComponentManager.java:266)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at org.wso2.carbon.kernel.internal.startupresolver.StartupComponentManager.notifySatisfiableComponents(StartupComponentManager.java:252)
        at org.wso2.carbon.kernel.internal.startupresolver.StartupOrderResolver$1.run(StartupOrderResolver.java:204)
        at java.util.TimerThread.mainLoop(Timer.java:555)
        at java.util.TimerThread.run(Timer.java:505)
Caused by: java.lang.NullPointerException
        at io.siddhi.extension.io.rabbitmq.source.RabbitMQSource.resume(RabbitMQSource.java:442)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at io.siddhi.core.SiddhiAppRuntimeImpl.lambda$restoreLastRevision$13(SiddhiAppRuntimeImpl.java:745)
        at java.util.concurrent.ConcurrentHashMap$ValuesView.forEach(ConcurrentHashMap.java:4707)
        at io.siddhi.core.SiddhiAppRuntimeImpl.restoreLastRevision(SiddhiAppRuntimeImpl.java:745)
        at org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorService.deploySiddhiApp(StreamProcessorService.java:212)
        at org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorDeployer.deploySiddhiQLFile(StreamProcessorDeployer.java:102)
        ... 14 more


### Steps to Reproduce

1. Set state.persistence enabled as true in deployment.yaml file in wso2 streaming integrator server.
2. Deploy a siddhi file which contain rabbitmq as source.
3.  Start the  wso2 streaming integrator server
Error logs :
 INFO {org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorService} - Periodic State persistence enabled. Restoring last persisted state of ReceiveRabbitmqInJSONFormat12
 ERROR {org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorDeployer} - java.lang.NullPointerException org.wso2.carbon.streaming.integrator.core.internal.exception.SiddhiAppDeploymentException: java.lang.NullPointerException
        at org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorDeployer.deploySiddhiQLFile(StreamProcessorDeployer.java:132)
        at org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorDeployer.deploy(StreamProcessorDeployer.java:358)
        at org.wso2.carbon.deployment.engine.internal.DeploymentEngine.lambda$deployArtifacts$0(DeploymentEngine.java:291)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at org.wso2.carbon.deployment.engine.internal.DeploymentEngine.deployArtifacts(DeploymentEngine.java:282)
        at org.wso2.carbon.deployment.engine.internal.RepositoryScanner.sweep(RepositoryScanner.java:112)
        at org.wso2.carbon.deployment.engine.internal.RepositoryScanner.scan(RepositoryScanner.java:68)
        at org.wso2.carbon.deployment.engine.internal.DeploymentEngine.start(DeploymentEngine.java:121)
        at org.wso2.carbon.deployment.engine.internal.DeploymentEngineListenerComponent.onAllRequiredCapabilitiesAvailable(DeploymentEngineListenerComponent.java:216)
        at org.wso2.carbon.kernel.internal.startupresolver.StartupComponentManager.lambda$notifySatisfiableComponents$7(StartupComponentManager.java:266)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at org.wso2.carbon.kernel.internal.startupresolver.StartupComponentManager.notifySatisfiableComponents(StartupComponentManager.java:252)
        at org.wso2.carbon.kernel.internal.startupresolver.StartupOrderResolver$1.run(StartupOrderResolver.java:204)
        at java.util.TimerThread.mainLoop(Timer.java:555)
        at java.util.TimerThread.run(Timer.java:505)
Caused by: java.lang.NullPointerException
        at io.siddhi.extension.io.rabbitmq.source.RabbitMQSource.resume(RabbitMQSource.java:442)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at io.siddhi.core.SiddhiAppRuntimeImpl.lambda$restoreLastRevision$13(SiddhiAppRuntimeImpl.java:745)
        at java.util.concurrent.ConcurrentHashMap$ValuesView.forEach(ConcurrentHashMap.java:4707)
        at io.siddhi.core.SiddhiAppRuntimeImpl.restoreLastRevision(SiddhiAppRuntimeImpl.java:745)
        at org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorService.deploySiddhiApp(StreamProcessorService.java:212)
        at org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorDeployer.deploySiddhiQLFile(StreamProcessorDeployer.java:102)
        ... 14 more


### Affected Component

SI

### Version

1.1.0,4.0.0

### Environment Details (with versions)

Windows Machine

### Relevant Log Output

```shell
ERROR {org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorDeployer} - java.lang.NullPointerException org.wso2.carbon.streaming.integrator.core.internal.exception.SiddhiAppDeploymentException: java.lang.NullPointerException
        at org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorDeployer.deploySiddhiQLFile(StreamProcessorDeployer.java:132)
        at org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorDeployer.deploy(StreamProcessorDeployer.java:358)
        at org.wso2.carbon.deployment.engine.internal.DeploymentEngine.lambda$deployArtifacts$0(DeploymentEngine.java:291)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at org.wso2.carbon.deployment.engine.internal.DeploymentEngine.deployArtifacts(DeploymentEngine.java:282)
        at org.wso2.carbon.deployment.engine.internal.RepositoryScanner.sweep(RepositoryScanner.java:112)
        at org.wso2.carbon.deployment.engine.internal.RepositoryScanner.scan(RepositoryScanner.java:68)
        at org.wso2.carbon.deployment.engine.internal.DeploymentEngine.start(DeploymentEngine.java:121)
        at org.wso2.carbon.deployment.engine.internal.DeploymentEngineListenerComponent.onAllRequiredCapabilitiesAvailable(DeploymentEngineListenerComponent.java:216)
        at org.wso2.carbon.kernel.internal.startupresolver.StartupComponentManager.lambda$notifySatisfiableComponents$7(StartupComponentManager.java:266)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at org.wso2.carbon.kernel.internal.startupresolver.StartupComponentManager.notifySatisfiableComponents(StartupComponentManager.java:252)
        at org.wso2.carbon.kernel.internal.startupresolver.StartupOrderResolver$1.run(StartupOrderResolver.java:204)
        at java.util.TimerThread.mainLoop(Timer.java:555)
        at java.util.TimerThread.run(Timer.java:505)
Caused by: java.lang.NullPointerException
        at io.siddhi.extension.io.rabbitmq.source.RabbitMQSource.resume(RabbitMQSource.java:442)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at io.siddhi.core.SiddhiAppRuntimeImpl.lambda$restoreLastRevision$13(SiddhiAppRuntimeImpl.java:745)
        at java.util.concurrent.ConcurrentHashMap$ValuesView.forEach(ConcurrentHashMap.java:4707)
        at io.siddhi.core.SiddhiAppRuntimeImpl.restoreLastRevision(SiddhiAppRuntimeImpl.java:745)
        at org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorService.deploySiddhiApp(StreamProcessorService.java:212)
        at org.wso2.carbon.streaming.integrator.core.internal.StreamProcessorDeployer.deploySiddhiQLFile(StreamProcessorDeployer.java:102)
        ... 14 more
```


### Related Issues

_No response_

### Suggested Labels

_No response_

--- a/component/src/main/java/io/siddhi/extension/io/rabbitmq/source/RabbitMQSource.java
+++ b/component/src/main/java/io/siddhi/extension/io/rabbitmq/source/RabbitMQSource.java
@@ -430,16 +430,20 @@ public class RabbitMQSource extends Source {
 
     @Override
     public void destroy() {
-
+        rabbitMQConsumer = null;
     }
 
     @Override
     public void pause() {
-       rabbitMQConsumer.pause();
+        if (rabbitMQConsumer != null) {
+            rabbitMQConsumer.pause();
+        }
     }
 
     @Override
     public void resume() {
-        rabbitMQConsumer.resume();
+        if (rabbitMQConsumer != null) {
+            rabbitMQConsumer.resume();
+        }
     }
 }

